lib/Basics/threads-win32.cpp

```diff
@@ -101,36 +101,36 @@ bool TRI_StartThread(TRI_thread_t* thread, char const* name,
 ////////////////////////////////////////////////////////////////////////////////
 
 int TRI_JoinThread(TRI_thread_t* thread) {
+  return TRI_JoinThreadWithTimeout(thread, INFINITE);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief waits for a thread to finish within the specified timeout (in ms).
+////////////////////////////////////////////////////////////////////////////////
+
+int TRI_JoinThreadWithTimeout(TRI_thread_t* thread, std::uint32_t timeout) {
```

Comment:
function behavior is undefined since `WAIT_ABANDONED` is not handled in `switch/case` statement and return statement is missing.

Reply:
`WAIT_ABANDONED` is not a valid return value when waiting on threads (it is only returned when you are trying to wait on an abandoned mutex).

Reply:
correct, but every single compiler/static analyzer will complain about that. We have to provide a default return value to avoid that.

Reply:
Good point about static analyzers! Fixed.

Reply:
Thanks!